diff --git a/config/commands.js b/config/commands.js
index dc8defd..3346982 100644
--- a/config/commands.js
+++ b/config/commands.js
@@ -1178,10 +1178,37 @@ var commands = exports.commands = {
 		// secret sysop command
 		room.add(target);
 	},
 
 	/*********************************************************
+	 * Custom commands
+	 *********************************************************/
+
+	stafflist: 'authlist',
+	authlist: function (target, room, user, connection) {
+		var rankLists = {};
+		for (var u in Users.usergroups) {
+			var rank = Users.usergroups[u][0];
+			var name = Users.usergroups[u].slice(1);
+			if (!rankLists[rank]) rankLists[rank] = [];
+			rankLists[rank].push(name);
+		}
+
+		var buffer = [];
+		Object.keys(rankLists).sort(function (a, b) {
+			return Config.groups.bySymbol[b].rank - Config.groups.bySymbol[a].rank;
+		}).forEach(function (r) {
+			buffer.push(Config.groups.bySymbol[r].name + "s (" + r + "):\n" + rankLists[r].sort().join(", "));
+		});
+
+		if (!buffer.length) {
+			buffer = "This server has no auth.";
+		}
+		connection.popup(buffer.join("\n\n"));
+	},
+
+	/*********************************************************
 	 * Help commands
 	 *********************************************************/
 
 	commands: 'help',
 	h: 'help',
