diff --git a/commands.js b/commands.js
index f1ceaed..8e8c72f 100644
--- a/commands.js
+++ b/commands.js
@@ -1597,11 +1597,10 @@ var commands = exports.commands = {
 		user.prepBattle(target, 'challenge', connection, function (result) {
 			if (result) user.makeChallenge(targetUser, target);
 		});
 	},
 
-	away: 'blockchallenges',
 	idle: 'blockchallenges',
 	blockchallenges: function (target, room, user) {
 		user.blockChallenges = true;
 		this.sendReply("You are now blocking all incoming challenge requests.");
 	},
diff --git a/config/commands.js b/config/commands.js
index be87fe0..9dffb04 100644
--- a/config/commands.js
+++ b/config/commands.js
@@ -1120,10 +1120,20 @@ var commands = exports.commands = {
 		// secret sysop command
 		room.add(target);
 	},
 
 	/*********************************************************
+	 * Custom commands
+	 *********************************************************/
+
+	away: function (target, room, user) {
+		user.away = !user.away;
+		user.updateIdentity();
+		this.sendReply("You are " + (user.away ? "now" : "no longer") + " away.");
+	},
+
+	/*********************************************************
 	 * Help commands
 	 *********************************************************/
 
 	commands: 'help',
 	h: 'help',
diff --git a/users.js b/users.js
index f8733d0..550c28f 100644
--- a/users.js
+++ b/users.js
@@ -382,26 +382,27 @@ var User = (function () {
 	User.prototype.popup = function (message) {
 		this.send('|popup|' + message.replace(/\n/g, '||'));
 	};
 	User.prototype.getIdentity = function (roomid) {
 		if (!roomid) roomid = 'lobby';
+		var name = this.name + (this.away ? " - \u0410\u051d\u0430\u0443" : "");
 		if (this.locked) {
-			return Config.lockedSymbol + this.name;
+			return Config.lockedSymbol + name;
 		}
 		if (this.mutedRooms[roomid]) {
-			return Config.mutedSymbol + this.name;
+			return Config.mutedSymbol + name;
 		}
 		var room = Rooms.rooms[roomid];
 		if (room.auth) {
 			if (room.auth[this.userid]) {
-				return room.auth[this.userid] + this.name;
+				return room.auth[this.userid] + name;
 			}
 			if (room.isPrivate) {
-				return Config.groups.default[room.type + 'Room'] + this.name;
+				return Config.groups.default[room.type + 'Room'] + name;
 			}
 		}
-		return this.group + this.name;
+		return this.group + name;
 	};
 	User.prototype.can = function (permission, target, room) {
 		if (this.hasSysopAccess()) return true;
 
 		var group = this.group;
