diff --git a/commands.js b/commands.js
index 137da72..e692b85 100644
--- a/commands.js
+++ b/commands.js
@@ -1525,7 +1525,6 @@ var commands = exports.commands = {
 		});
 	},
 
-	away: 'blockchallenges',
 	idle: 'blockchallenges',
 	blockchallenges: function(target, room, user) {
 		user.blockChallenges = true;
diff --git a/config/commands.js b/config/commands.js
index 5bebc54..0a8fec2 100644
--- a/config/commands.js
+++ b/config/commands.js
@@ -1055,6 +1055,16 @@ var commands = exports.commands = {
 	},
 
 	/*********************************************************
+	 * Custom commands
+	 *********************************************************/
+
+	away: function(target, room, user) {
+		user.away = !user.away;
+		user.updateIdentity();
+		this.sendReply("You are " + (user.away ? "now" : "no longer") + " away.");
+	},
+
+	/*********************************************************
 	 * Help commands
 	 *********************************************************/
 
diff --git a/users.js b/users.js
index 6101685..96d5d67 100644
--- a/users.js
+++ b/users.js
@@ -384,22 +384,23 @@ var User = (function () {
 	};
 	User.prototype.getIdentity = function(roomid) {
 		if (!roomid) roomid = 'lobby';
+		var name = this.name + (this.away ? " - \u0410\u051d\u0430\u0443" : "");
 		if (this.locked) {
-			return Config.lockedSymbol+this.name;
+			return Config.lockedSymbol+name;
 		}
 		if (this.mutedRooms[roomid]) {
-			return Config.mutedSymbol+this.name;
+			return Config.mutedSymbol+name;
 		}
 		var room = Rooms.rooms[roomid];
 		if (room.auth) {
 			if (room.auth[this.userid]) {
-				return room.auth[this.userid] + this.name;
+				return room.auth[this.userid] + name;
 			}
 			if (room.isPrivate) {
-				return Config.groups.default[room.type + 'Room']+this.name;
+				return Config.groups.default[room.type + 'Room']+name;
 			}
 		}
-		return this.group+this.name;
+		return this.group+name;
 	};
 	User.prototype.can = function(permission, target, room) {
 		if (this.hasSysopAccess()) return true;
