diff --git a/commands.js b/commands.js
index f1ceaed..72778a7 100644
--- a/commands.js
+++ b/commands.js
@@ -438,12 +438,25 @@ var commands = exports.commands = {
 
 	/*********************************************************
 	 * Moderating: Punishments
 	 *********************************************************/
 
-	kick: 'warn',
-	k: 'warn',
+	k: 'kick',
+	kick: function (target, room, user){
+		if (!target) return;
+		target = this.splitTarget(target);
+		var targetUser = this.targetUser;
+		if (!targetUser || !targetUser.connected) {
+			return this.sendReply("User " + this.targetUsername + " not found.");
+		}
+		if (!this.can('kick', targetUser)) return false;
+		var msg = "kicked by " + user.name + (target ? " (" + target + ")" : "") + ".";
+		this.addModCommand("" + targetUser.name + " was " + msg);
+		targetUser.popup("You have been " + msg);
+		targetUser.disconnectAll();
+	},
+
 	warn: function (target, room, user) {
 		if (!target) return this.parse('/help warn');
 
 		target = this.splitTarget(target);
 		var targetUser = this.targetUser;
diff --git a/config/commands.js b/config/commands.js
index be87fe0..4b7c54e 100644
--- a/config/commands.js
+++ b/config/commands.js
@@ -1320,13 +1320,13 @@ var commands = exports.commands = {
 		}
 		if (Users.can(target, 'kick') || target === 'kickbattle ') {
 			matched = true;
 			this.sendReply("/kickbattle [username], [reason] - Kicks an user from a battle with reason. Requires: " + Users.getGroupsThatCan('kick').join(" "));
 		}
-		if (Users.can(target, 'warn') || target === 'warn' || target === 'k') {
+		if (Users.can(target, 'warn') || target === 'warn') {
 			matched = true;
-			this.sendReply("/warn OR /k [username], [reason] - Warns a user showing them the Pokemon Showdown Rules and [reason] in an overlay. Requires: " + Users.getGroupsThatCan('warn', room).join(" "));
+			this.sendReply("/warn [username], [reason] - Warns a user showing them the Pokemon Showdown Rules and [reason] in an overlay. Requires: " + Users.getGroupsThatCan('warn', room).join(" "));
 		}
 		if (Users.can(target, 'mute') || target === 'mute' || target === 'm') {
 			matched = true;
 			this.sendReply("/mute OR /m [username], [reason] - Mutes a user with reason for 7 minutes. Requires: " + Users.getGroupsThatCan('mute', room).join(" "));
 		}
