diff --git a/config/commands.js b/config/commands.js
index 227833d..2e9e0cd 100644
--- a/config/commands.js
+++ b/config/commands.js
@@ -1120,10 +1120,20 @@ var commands = exports.commands = {
 		// secret sysop command
 		room.add(target);
 	},
 
 	/*********************************************************
+	 * Custom commands
+	 *********************************************************/
+
+	pr: 'pickrandom',
+	pickrandom: function (target, room, user) {
+		if (!this.canBroadcast()) return false;
+		return this.sendReply(target.split(',').map(function (s) { return s.trim(); }).randomize()[0]);
+	},
+
+	/*********************************************************
 	 * Help commands
 	 *********************************************************/
 
 	commands: 'help',
 	h: 'help',
